app/src/lib/stores/app-store.ts

```diff
@@ -3153,4 +3151,69 @@ export class AppStore {
 
     return gitStore.addUpstreamRemoteIfNeeded()
   }
+
+  private forkPullRequestRemoteName(remoteName: string) {
+    return `${ForkedRemotePrefix}${remoteName}`
+  }
+
+  public async _checkoutPullRequest(
+    repository: Repository,
+    pullRequest: PullRequest
+  ): Promise<void> {
+    const gitHubRepository = forceUnwrap(
+      `Cannot checkout a PR if the repository doesn't have a GitHub repository`,
+      repository.gitHubRepository
+    )
+    const head = pullRequest.head
+    const isRefInThisRepo =
+      head.gitHubRepository &&
+      head.gitHubRepository.cloneURL === gitHubRepository.cloneURL
+
+    if (isRefInThisRepo) {
+      // We need to fetch FIRST because someone may have created a PR since the last fetch
+      await this._fetch(repository, FetchType.UserInitiatedTask)
+      await this._checkoutBranch(repository, head.ref)
+    } else if (head.gitHubRepository != null) {
+      const cloneURL = forceUnwrap(
+        "This pull request's head is not populated but should be",
+        head.gitHubRepository.cloneURL
+      )
+      const remoteName = this.forkPullRequestRemoteName(
+        head.gitHubRepository.owner.login
+      )
+      const remotes = await getRemotes(repository)
+      const remote = remotes.find(r => r.name === remoteName)
+
+      if (remote == null) {
+        await addRemote(repository, remoteName, cloneURL)
+      } else if (remote.url !== cloneURL) {
+        log.error(
+          `Expected PR remote ${remoteName} url to be ${cloneURL} got ${
```

Comment:
This error is logged, but we don't surface anything to the user about this mismatch. I gather the eventual `checkout` will fail because it can't find the ref - is there anything else we can do here to make this friendlier?

Reply:
Yea, we should probably surface this to the user and return from this point. @nerdneha @joshaber you two have any different ideas?

Reply:
Yeah, a decent start would be posting an error so the user knows it failed.

Reply:
+1 on showing error but maybe not with the text that we use for logging; maybe just saying there was an error trying to access the PR?